apps/entity-service/src/entities/processors/site.processor.ts

```diff
@@ -1,24 +1,161 @@
+      });
+    }
+
+    for (const term of ["name", "projectName"]) {
+      if (query[term] != null) builder.where({ [term]: query[term] });
+    }
+    if (query.search != null) {
+      builder.where({ name: { [Op.like]: `%${query.search}%` } });
+    }
+
+    return { models: await builder.execute(), paginationTotal: await builder.paginationTotal() };
+  }
+
+  async addFullDto(document: DocumentBuilder, site: Site): Promise<void> {
+    const siteId = site.id;
+
+    const approvedSiteReportsQuery = SiteReport.approvedIdsSubquery({ val: siteId });
+    const seedsPlantedCount = (await Seeding.visible().siteReports(approvedSiteReportsQuery).sum("amount")) ?? 0;
+    const treesPlantedCount =
+      (await TreeSpecies.visible().collection("tree-planted").siteReports(approvedSiteReportsQuery).sum("amount")) ?? 0;
+
+    const hasBeenSubmittedSiteReports = await SiteReport.hasBeenSubmitted()
+      .sites([siteId])
+      .findAll({ attributes: ["id", "siteId", "numTreesRegenerating"] });
+
+    const approvedSiteReports = await SiteReport.approved()
+      .sites([siteId])
+      .findAll({ attributes: ["id", "siteId", "numTreesRegenerating"] });
+
+    const regeneratedTreesCount = sumBy(hasBeenSubmittedSiteReports, "numTreesRegenerating");
+    const approvedRegeneratedTreesCount = sumBy(approvedSiteReports, "numTreesRegenerating");
+
+    const props: AdditionalSiteFullProps = {
+      totalHectaresRestoredSum: (await SitePolygon.approved().sites([site.uuid]).sum("calcArea")) ?? 0,
+      workdayCount: (await this.getWorkdayCount(siteId)) ?? 0,
+      combinedWorkdayCount:
+        (await this.getWorkdayCount(siteId, true)) + (await this.getSelfReportedWorkdayCount(siteId, true)),
+      siteReportsTotal: await this.getTotalSiteReports(siteId),
+      seedsPlantedCount,
+      overdueSiteReportsTotal: await this.getTotalOverdueReports(siteId),
+      selfReportedWorkdayCount: await this.getSelfReportedWorkdayCount(siteId, true),
+      treesPlantedCount,
+      regeneratedTreesCount,
+      approvedRegeneratedTreesCount,
+
+      ...(this.entitiesService.mapMediaCollection(await Media.site(siteId).findAll(), Site.MEDIA) as SiteMedia)
+    };
+
+    document.addData(site.uuid, new SiteFullDto(site, props));
+  }
+
+  async addLightDto(document: DocumentBuilder, site: Site) {
+    document.addData(site.uuid, new SiteLightDto(site));
+  }
+
+  protected async getWorkdayCount(siteId: number, useDemographicsCutoff = false) {
+    const dueAfter = useDemographicsCutoff ? Demographic.DEMOGRAPHIC_COUNT_CUTOFF : undefined;
+
+    const siteReportIds = SiteReport.approvedIdsSubquery({ val: siteId }, { dueAfter });
+    const siteReportWorkdays = Demographic.idsSubquery(
+      siteReportIds,
+      SiteReport.LARAVEL_TYPE,
+      Demographic.WORKDAYS_TYPE
+    );
+
+    return (
+      (await DemographicEntry.gender().sum("amount", {
+        where: {
+          demographicId: {
+            [Op.or]: [{ [Op.in]: siteReportWorkdays }]
```

Comment:
When there's only one, there's no reason to wrap it in an `or`. This can be simply
```
demographicId: { [Op.in]: siteReportWorkdays }
```